frappe/core/doctype/system_settings/system_settings.js

```diff
@@ -27,3 +42,89 @@ frappe.ui.form.on("System Settings", "enable_two_factor_auth", function(frm) {
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = null;
+
+    }
+    cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = function () {
+
+        if (frm.doc.new_password_encryption == frm.doc.confirm_password_encryption) {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: #08c708"
+
+        } else {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: red"
+
+        }
+
+    }
+
+});
+
+frappe.ui.form.on("System Settings", "change_encryption_password", function (frm) {
+    if (frm.doc.enable_backup_encryption && frm.doc.change_encryption_password) {
+        frm.fields_dict.new_password_encryption.df.reqd = 1;
+        frm.fields_dict.new_password_encryption.df.hidden = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 1;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.new_password_encryption.input_area.children[0].value = "******";
+        cur_frm.fields_dict.confirm_password_encryption.input_area.children[0].value = "******";
+
+
+    } else {
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.fields_dict.new_password_encryption.df.hidden = 1;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 0;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 1;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = null;
+    }
+    cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = function () {
+
+        if (frm.doc.new_password_encryption == frm.doc.confirm_password_encryption) {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: #08c708"
+
+        } else {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: red"
+
+        }
+
+    }
+
+
+});
+
+frappe.ui.form.on("System Settings", "validate", function (frm) {
+    if (frm.doc.new_password_encryption != frm.doc.confirm_password_encryption && frm.doc.enable_backup_encryption) {
+        frappe.msgprint("Password encrytion is not matched with the confirmed password.");
+        frappe.validated = False;
+    } else if (frm.doc.new_password_encryption == frm.doc.confirm_password_encryption && frm.doc.enable_backup_encryption) {
+        frm.doc.password_encryption = "******"
+        frappe.msgprint("Password is saved and cannot be retrieved/viewed.");
```

Comment:
Isn't this the default behavior of password fields?

Reply:
we removed this line now ```frm.doc.password_encryption = "******"```

Reply:
Hi, any updates for this? thanks.